src/index.js

```diff
@@ -0,0 +1,101 @@
+const skyLeft = {
+  Sunny: '☀️🌤🌞🌤☀️',
+  Cloudy: '🌫☁️🌥⛅️🌤',
+  Rainy: '🌈⛈🌧🌧🌦',
+  Snowy: '🌨❄️🌨❄️🌨',
+};
+
+const skyRight = {
+  Sunny: '☀️🌤🌞🌤☀️',
+  Cloudy: '🌤⛅️🌥☁️🌫',
+  Rainy: '🌦🌧🌧⛈💧',
+  Snowy: '🌨❄️🌨❄️🌨',
+};
+
+const skyColor = {
+  Sunny: 'skyblue',
+  Cloudy: 'lightgrey',
+  Rainy: 'lightblue',
+  Snowy: 'lightsteelblue',
+};
+
+const skySelector = document.getElementById('sky__selector');
+const leftSky = document.getElementById('sky__left');
+const rightSky = document.getElementById('sky__right');
+
+const defaultSky = skySelector.options[skySelector.selectedIndex].text;
+
+leftSky.textContent = skyLeft[defaultSky];
+rightSky.textContent = skyRight[defaultSky];
+
+const backgroundColor = document.getElementsByClassName('weather__garden')[0];
+
+backgroundColor.style.backgroundColor = skyColor['Sunny'];
+
+skySelector.addEventListener('change', function () {
+  const currentSky = skySelector.options[skySelector.selectedIndex].text;
+  leftSky.textContent = skyLeft[currentSky];
+  rightSky.textContent = skyRight[currentSky];
+  backgroundColor.style.backgroundColor = skyColor[currentSky];
+});
+
+const state = {
+  temperature: 72,
+};
+
+const increaseButton = document.getElementById('increase');
+const decreaseButton = document.getElementById('decrease');
+const temperature = document.getElementById('temp');
+
+const garden = document.getElementById('garden');
+
+temperature.textContent = state.temperature;
+
+const updateGarden = function () {
+  if (state.temperature > 79) {
+    garden.textContent = '🌵 🐪 🏜 🐍 🦂 🌵 🌵 🐍 🏜 🐫 🦂 🌵';
+    temperature.style.color = 'red';
+  }
+  if (state.temperature > 69 && state.temperature < 80) {
```

Comment:
If these conditions were all connected using `else if`, then we could get away with only checking one end of the temperature ranges. Consider
```js
  if (state.temperature > 79) {
    ...
  } else if (state.temperature > 69) {
    ...
  }
```
Since the second condition will only be check if the first fails, we know that the temperature must be less than or equal to 79.

Reply:
Even cooler would be to set up a data structure similar to what you did for the sky that would allow these checks to be data-driven. Since checking a range of values is involved, we can't do a simple dictionary lookup, but instead, we could make a list of records to iterate through, looking for the record that applies for the current temperature.

Consider the above logic and how similar all the branches are. We look for whether the temperature is within a range, and pick a color and landscape accordingly. So we could set up something like
```js
  const tempRecords = [
    { lowerBound: 79, color: "red", landscape: "🌵 🐪 🏜 🐍 🦂 🌵 🌵 🐍 🏜 🐫 🦂 🌵"},
    { lowerBound: 69, color: "other color", landscape: "other landscape"},
    ...
  ];
```

For the final case, we might set the `lowerBound` to a very low value, or to `null` and write a special case to check it. But for everything else, we could simply iterate through the list and find the first record that has a lower bound _lower_  than our temperature, then use it as the source of picking the color and landscape!